pkg/auth/oauth/registry/timeoutvalidator.go

```diff
@@ -0,0 +1,298 @@
+package registry
+
+import (
+	"errors"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/btree"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/openshift/origin/pkg/oauth/apis/oauth"
+	oauthclient "github.com/openshift/origin/pkg/oauth/generated/internalclientset/typed/oauth/internalversion"
+	oauthclientlister "github.com/openshift/origin/pkg/oauth/generated/listers/oauth/internalversion"
+	"github.com/openshift/origin/pkg/user/apis/user"
+)
+
+var ErrTimedout = errors.New("token timed out")
```

Comment:
nit: I believe this can be made private.

Reply:
ok